provider/identityprovider/htpasswd.go

```diff
@@ -86,3 +89,24 @@ func CreateHTPasswdIDPBuilder(ctx context.Context, state *HTPasswdIdentityProvid
 	builder.Users(userListBuilder)
 	return builder
 }
+
+func uniqueUsernameValidator() validator.List {
+	return attrvalidators.NewListValidator("userlist unique username", func(ctx context.Context, req validator.ListRequest, resp *validator.ListResponse) {
+		usersList := req.ConfigValue
+		htusers := []HTPasswdUser{}
+		err := usersList.ElementsAs(ctx, &htusers, true)
+		if err != nil {
+			resp.Diagnostics.AddAttributeError(req.Path, "Invalid list conversion", "Failed to parse userlist")
+			return
+		}
+		usernames := make(map[string]bool)
```

Comment:
I think better to use here array (not map) and for each name check if already exists in the array

Reply:
Ok, never mind, this is good

Reply:
:)
I do think that O(1) in finding in this case is "less expensive" than O(n) of memory...